code/modules/projectiles/projectile/homing_projectiles.dm

```diff
@@ -50,3 +50,43 @@
 		var/mob/living/carbon/human/H = target
 		var/obj/item/organ/external/E = pick(H.bodyparts)
 		E.add_embedded_object(T)
+
+/obj/item/projectile/homing/magic/homing_fireball
+	name = "greater bolt of fireball"
+	icon_state = "fireball"
+	damage = 20
+	damage_type = BRUTE
+	nodamage = FALSE
+
+	//explosion values
+	var/exp_devastate = 0
+	var/exp_heavy = 1
+	var/exp_light = 3
+	var/exp_flash = 4
+	var/exp_fire = 3
+
+/obj/item/projectile/homing/magic/homing_fireball/Range()
+	var/turf/T1 = get_step(src,turn(dir, -45))
+	var/turf/T2 = get_step(src,turn(dir, 45))
+	var/turf/T3 = get_step(src,dir)
+	var/mob/living/L = locate(/mob/living) in T1 //if there's a mob alive in our front right diagonal, we hit it.
+	if(L && L.stat != DEAD)
+		Bump(L) //Magic Bullet #teachthecontroversy
+		return
+	L = locate(/mob/living) in T2
+	if(L && L.stat != DEAD)
+		Bump(L)
+		return
+	L = locate(/mob/living) in T3
+	if(L && L.stat != DEAD)
+		Bump(L)
+		return
+	..()
```

Comment:
```suggestion
	var/mob/living/mob_to_fireball = locate(/mob/living) in get_step(src, dir) // If there's a mob alive in the three tiles right in front of us, we slam into them
	if(mob_to_fireball && mob_to_fireball.stat != DEAD)
		Bump(mob_to_fireball) // Magic Boolet #teachthecontroversy
		return
	mob_to_fireball = locate(/mob/living) in get_step(src, turn(dir, 45))
	if(mob_to_fireball && mob_to_fireball.stat != DEAD)
		Bump(mob_to_fireball)
		return
	mob_to_fireball = locate(/mob/living) in get_step(src, turn(dir, 45))
	if(mob_to_fireball && mob_to_fireball.stat != DEAD)
		Bump(mob_to_fireball)
		return
	..()
```
we could cut down on this proc even more by using a list

Reply:
I'm gonna get rid of this and just make it explode

Reply:
These bumps never did anything anyways, removing